frontend/app/module/input/PollingStation/AbortDataEntryControl.test.tsx

```diff
@@ -0,0 +1,102 @@
+
+import { AbortDataEntryControl } from "./AbortDataEntryControl.tsx";
+
+const mockNavigate = vi.fn();
+
+const renderAbortDataEntryControl = () => {
+  render(
+    <ElectionProvider electionId={1}>
+      <PollingStationFormController
+        election={electionMock}
+        pollingStationId={pollingStationMock.id}
+        entryNumber={1}
+      >
+        <AbortDataEntryControl />
+      </PollingStationFormController>
+    </ElectionProvider>,
+  );
+};
+
+describe("Test AbortDataEntryControl", () => {
+  beforeEach(() => {
+    overrideOnce("get", "/api/elections/1", 200, electionMockResponse);
+    vi.spyOn(router, "useNavigate").mockImplementation(() => mockNavigate);
+  });
+
+  afterEach(() => {
+    vi.clearAllMocks();
+    server.restoreHandlers();
+    server.events.removeAllListeners();
+  });
+
+  test("renders and toggles the modal", async () => {
+    // render the abort button, then click it to open the modal
+    renderAbortDataEntryControl();
+    const user = userEvent.setup();
+
+    // click the abort button to open the modal
+    const abortButton = await screen.findByRole("button", { name: "Invoer afbreken" });
+    await user.click(abortButton);
+
+    // check that the modal is open
+    expect(screen.getByText("Wat wil je doen met je invoer?")).toBeInTheDocument();
+  });
+
+  test("saves the form state and navigates on save", async () => {
+    // render the abort button
+    renderAbortDataEntryControl();
+    const user = userEvent.setup();
+
+    // click the abort button to open the modal
+    const abortButton = await screen.findByRole("button", { name: "Invoer afbreken" });
+    await user.click(abortButton);
+
+    // click the save button in the modal
+    fireEvent.click(screen.getByText("Invoer bewaren"));
+
+    // check that the user is navigated back to the input page
+    expect(mockNavigate).toHaveBeenCalledWith(expect.stringContaining("/input"));
+  });
+
+  test("deletes the data entry and navigates on delete", async () => {
+    // render the abort button, then click it to open the modal
+    renderAbortDataEntryControl();
+    const user = userEvent.setup();
+
+    // click the abort button to open the modal
+    const abortButton = await screen.findByRole("button", { name: "Invoer afbreken" });
+    await user.click(abortButton);
+
+    // set up a listener to check if the delete request is made
+    let deleteRequestMade = false;
+    overrideOnce("delete", "/api/polling_stations/1/data_entries/1", 204, null);
+    server.events.on("request:start", ({ request }) => {
+      if (
+        request.method === "DELETE" &&
+        request.url === "http://testhost/api/polling_stations/1/data_entries/1"
+      ) {
+        deleteRequestMade = true;
+      }
+    });
+
+    // click the delete button in the modal
+    fireEvent.click(screen.getByText("Niet bewaren"));
+
+    // check that the delete request was made
+    expect(deleteRequestMade).toBe(true);
```

Comment:
```suggestion
    // set up a listener to check if the delete request is made
    const spy = vi.spyOn(global, "fetch");
    overrideOnce("delete", "/api/polling_stations/1/data_entries/1", 204, null);

    // click the delete button in the modal
    fireEvent.click(screen.getByText("Niet bewaren"));

    // check that the delete request was made
    expect(spy).toHaveBeenCalled();
    const { url, method, body } = getUrlMethodAndBody(spy.mock.calls);
    expect(url).toEqual("http://testhost/api/polling_stations/1/data_entries/1");
    expect(method).toEqual("DELETE");
    expect(body).toBeUndefined();
```

Reply:
Why do you prefer a `fetch` spy over the MSW event listener?

Reply:
Because that's how we do it in other tests.

MSW does recommend [using events if you can't assert on the component's behavior](https://mswjs.io/docs/best-practices/avoid-request-assertions/#exceptions), so that's an argument in favor of events. And it would allows us to get rid of `getUrlMethodAndBody()`.

So I'm now convinced of using the MSW event listener.

I would still prefer to replace `deleteRequestMade` with `url` and `method` and `expect()` on those separately, so that if the test fails, it's clearer why it fails.

Reply:
> I would still prefer to replace `deleteRequestMade` with `url` and `method` and `expect()` on those separately, so that if the test fails, it's clearer why it fails.

Changed in 0a59ba3. `deleteRequestMade` is still needed to know that the listener was called at all.

Reply:
I should have elaborated more.

With the `expect()` in the `server.events.on()` the test does not fail very nicely. You get:
- an `AssertionError` on `expect(deleteRequestMade).toBe(true);`
- a `vitest-fail-on-console` error that tells you how the url and/or method do not match, which is caused by
- the `PollingStationFormController.tsx` throwing an error on line 177: "Failed to delete data entry"

So what I was thinking was to get rid of the `deleteRequestMade` and initialize, assign and assert on `url` and `method` in the same way as is done now for the `deleteRequestMade`.

Reply:
Changed in a01fecdf200011799b942229c0384cce26e6ffa8. This test will break if other requests are made after this one; the original test didn't have that issue.

Reply:
I'd say that's a feature.